src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -427,13 +427,88 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     return true;
 }
 
+// Convert the return kind that was encoded by RyuJIT to the
+// value that CoreRT runtime can understand and support.
+GCRefKind GetGcRefKind(ReturnKind returnKind)
+{
+    static_assert((GCRefKind)ReturnKind::RT_Scalar == GCRK_Scalar, "ReturnKind::RT_Scalar does not match GCRK_Scalar");
```

Comment:
I just followed the pattern that we have in RhCodeMan (https://github.com/dotnet/corert/blob/master/src/Native/Runtime/RHCodeMan.cpp#L1028). But I do agree with you that either the static_asserts or the switch statement is not necessary. I will remove the switch statement to make the code a bit shorter and to save a few instructions.